crowbar_framework/app/controllers/api/restart_management_controller.rb

```diff
@@ -81,19 +82,35 @@ def restarts_get
 
   def restarts_post
     params.require(:node)
-    params.require(:service)
 
     node_name = params[:node]
+    cookbook = params[:cookbook]
     service = params[:service]
 
     node = get_node_or_raise(node_name)
 
-    managed_cookbooks.each do |cookbook|
-      next unless node.key? :crowbar_wall
-      next unless node[:crowbar_wall].key? :requires_restart
-      next unless node[:crowbar_wall][:requires_restart].key? cookbook
-      next unless node[:crowbar_wall][:requires_restart][cookbook].key? service
-      node[:crowbar_wall][:requires_restart][cookbook].delete(service)
+    raise Crowbar::Error::NotFound unless
+      node.key?(:crowbar_wall) && node[:crowbar_wall].key?(:requires_restart)
+
+    if cookbook.nil?
+      node[:crowbar_wall].delete(:requires_restart)
+      Rails.logger.info("restart_management : node #{node_name} cleaned")
+    else
+      raise Crowbar::Error::NotFound unless managed_cookbooks.include? cookbook
+      raise Crowbar::Error::NotFound unless node[:crowbar_wall][:requires_restart].key? cookbook
```

Comment:
Same here: do we really want to raise an error (for the second line, the first one is fine).